### PR TITLE
Crac import in two steps

### PR DESCRIPTION
**Pull request by baptsegu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

_New feature_: new API layer to be able to import Crac in two steps.

- first step: from a file, or an input  steam, import a `NativeCrac` (java object). The `NativeCrac` is a raw translation of file, based for instance on the generated classes of its xsd. It contains all the information present in the file.

- second step: create a `Crac`, from a `NativeCrac` and a `Network`.

The benefits of doing this in two steps are:
- to keep all the raw information of the input file in the NativeCrac object
- to be able to have a network when interpreting the `NativeCrac` into a `Crac` object, and therefore be able to perform some operation with the network (e.g. read IST, check the type of the network element, do a cross-quality check with the network, ...), which are currently done in less intuitive methods such as `synchronize`
- to know how the `NativeCrac` information have been mapped in the `Crac`, thanks to the `CracCreationContext` object. This object can be usefull to reverse some information when exporting the `Crac`

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

Not yet, as the "old importer' still exist.

### CODE REVIEW

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public interface CracCreator<T extends NativeCrac, S extends CracCreationContext> {
+
+    /**
+     * Get a unique identifier of the native Crac format handled by the CracCreator.
+     */
+    String getNativeCracFormat();
+
+    /**
+     * Create a Crac object from a NativeCrac and a Network.
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreator.java:31]: done :)

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+public final class CracCreators {
+
+    private static final Supplier<List<CracCreator>> CRAC_CREATORS
+        = Suppliers.memoize(() -> new ServiceLoaderCache<>(CracCreator.class).getServices())::get;
+
+    private CracCreators() {
+    }
+
+    /**
+     * Flexible method to create a Crac from a native CRAC and a network, whatever the format of the
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java:39]: done

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (Objects.isNull(creator)) {
+            throw new FaraoException(String.format("No CracCreator found for format %s", nativeCrac.getFormat()));
+        }
+
+        return creator.createCrac(nativeCrac, network, offsetDateTime);
+    }
+
+    /**
+     * Flexible method to import a Crac from a native CRAC file and a network, whatever the format of the
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java:57]: done

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @param offsetDateTime timestamp for which the Crac is creator (null values might be accepted by some creators)
+     * @return the created {@link NativeCrac} object
+     */
+    public static CracCreationContext importAndCreateCrac(Path nativeCracPath, Network network, OffsetDateTime offsetDateTime) {
+        NativeCrac nativeCrac = NativeCracImporters.importData(nativeCracPath);
+        return createCrac(nativeCrac, network, offsetDateTime);
+    }
+
+    /**
+     * Flexible method to import a Crac from a native CRAC file and a network, whatever the format of the
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/CracCreators.java:70]: done

File: data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/std_creation_context/BranchCnecCreationContext.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Get a boolean indicating whether the native CNEC is monitored in preventive, or after a contingency
+     * Underlying assumption: one native CNEC cannot be monitored in preventive and after a contingency
+     */
+    boolean isBaseCase();
+
+    /**
+     * Get the id of the native CNEC contingency
+     * Underlying assumption: one native CNEC definition do not cover several contingencies
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/src/main/java/com/farao_community/farao/data/crac_creator_api/std_creation_context/BranchCnecCreationContext.java:37]: :+1:

File: data/crac/crac-creator-api/pom.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>farao-native-crac-io-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>farao-crac-api</artifactId>
```
[the remark this answers is not in the document]
Comment (baptsegu) [reply data/crac/crac-creator-api/pom.xml]: corrected

**Review by pet-mit — APPROVED** [review]

**Pull request merged**